WooCommerce-Wear/build.gradle

```
@@ -77,19 +79,13 @@ android {
         }
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
-    }
-    kotlinOptions {
-        jvmTarget = '1.8'
+        sourceCompatibility JvmTarget.fromTarget(libs.versions.java.get()).target
```

I suggest simplifying this to:

```suggestion
        sourceCompatibility libs.versions.java.get()
```

---

Good suggestion @wzieba ! 👍 

Frankly, I got/stole this idea from [PCAndroid](https://github.com/Automattic/pocket-casts-android/blob/1585aad387eb28883f61ba8d48224c81883e3080/build.gradle.kts#L129), but which is on Kotlin DSL for Gradle already. Thus, I thought it will be a good idea to use that now, versus just using `libs.versions.java.get()` directly. The only drawback being the boilerplate code with wrapping everything into `JvmTarget.fromTarget(...).target`. My thinking was:

1. **Type Safety**: The `JvmTarget.fromTarget()` method ensures that the provided version string is a valid JVM target. It will throw an exception if an invalid version is provided, catching potential errors early.
2. **Consistency with Kotlin DSL**: When using the Kotlin DSL for Gradle, JvmTarget integrates more naturally with the strongly-typed nature of Kotlin. `jvmTarget.set(javaTarget)` versus `jvmTarget.set(libs.versions.java.get())` which will not be allowed.

I don't feel strongly about it, so if you insist on changing that to `libs.versions.java.get()` after reading the above, I'll change it everywhere.

---

> Type Safety

I don't think type safety is important here: this is simply a version of Java, so if it's incorrect, then it'll be caught at the same as if you wrap it with `JvmTarget`. 

> Consistency with Kotlin DSL

Sorry, I'm not sure if I follow - as we don't use Kotlin DSL here, I don't think we should focus on Kotlin DSL.

It's a minor, so I don't insist. I just thought having less code would be simply better in this case.

> PS: Same applies to [JP/WPAndroid](https://github.com/wordpress-mobile/WordPress-Android), which already got merged into trunk with JvmTarget.fromTarget(libs.versions.java.get()).target. I was hoping this would have been discussed there instead

I can assure you I didn't do this on purpose! I simply spotted this improvement in this PR, not the other one.

---

> I can assure you I didn't do this on purpose!

Wasn't implying anything! 🤣

> I don't think type safety is important here: this is simply a version of Java, so if it's incorrect, then it'll be caught at the same as if you wrap it with JvmTarget.

👍 

> Sorry, I'm not sure if I follow - as we don't use Kotlin DSL here, I don't think we should focus on Kotlin DSL.

Yea, I was trying to be smart about it and prepare in advance, not worth it (I guess). 👍 

> It's a minor, so I don't insist. I just thought having less code would be simply better in this case.

Let's have less code then! 😉 (yes, I know how you feel about 😉 -> 😜 🤣)

:onit:

---

This is now done: a65218cb2ca83dcbc560912c1731e64d36a5d793